src/pages/index.js

```diff
@@ -3,95 +3,157 @@ import Card from "/src/components/Card.js";
-
-const editModal = document.getElementById("edit-modal");
-const newCardModal = document.getElementById("new-card-modal");
-const previewImageModal = document.querySelector("#preview-image-modal");
-
-const previewImage = previewImageModal.querySelector(".preview-image");
-const previewImageText = previewImageModal.querySelector(".image-description");
-
-const profileEditBtn = document.querySelector(".profile__edit-button");
-const newCardBtn = document.querySelector(".profile__plus-button");
-
-const formValidators = {};
-
-const enableValidation = (config) => {
-  const formList = [...document.querySelectorAll(config.formSelector)];
-  formList.forEach((form) => {
-    const validator = new FormValidator(config, form);
-    const formName = form.getAttribute("name");
+import * as constants from "/src/utils/constants.js";
+import Api from "/src/components/Api.js";
+// User Data
+const userInfo = new UserInfo({
+  userName: constants.profileName,
+  userDescription: constants.profileDescription,
+  userProfileImage: constants.profileAvatar,
+});
+// Api Instantiation
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "9fbfab1b-69d6-43a9-b085-60d57b074309",
+    "Content-Type": "application/json",
+  },
+});
 
-    formValidators[formName] = validator;
-    validator.enableValidation();
-  });
-};
-const createCard = (data) => {
-  const cardElement = new Card({
-    data: data,
-    cardSelector: cardTemplate,
-    handleCardClick: () => {
-      popupImage.open(data);
+Promise.all([api.getUserInformation(), api.getInitialCards()]).then(
+  ([userData, initialCards]) => {
+    userInfo.setUserInfo(userData);
+    userInfo.setUserProfileImage(userData);
+    const apiCardSection = new Section(
+      {
+        items: initialCards,
+        renderer: addCard,
+      },
+      constants.cardGallery
+    );
+    apiCardSection.renderItems();
+  }
+);
+// Create Card + Add Card Functions
+const createCard = (cardData) => {
+  const cardElement = new Card(
+    cardData,
+    {
+      handleCardClick: () => popupImage.open(cardData),
+      handleDeleteClick: () => {
+        deleteModalNew.open();
+        deleteModalNew.submitForm(() => {
+          api
+            .deleteCard(cardData._id)
+            .then(() => {
+              deleteModalNew.close();
+              cardElement.removeCard();
+            })
+            .catch((err) => console.error(err));
+        });
+      },
+      handleLikeClick: (card) => {
+        // debugger;
+        if (!cardData.isLiked) {
+          api
+            .addLike(card)
+            .then((res) => {
+              cardElement.setLike(res);
+              console.log("add");
+              console.log(res.isLiked);
+            })
+            .catch((err) => console.error(err));
+        } else {
+          api
+            .deleteLike(card)
+            .then((res) => {
+              cardElement.setLike(res.isLiked);
```

Comment:
* I can't dislike **new** cards. The black heart doesn't get white

Reply:
Thanks for your feedback. I'm not sure how to get the functionality of the like button correct. It works on initial load and on refresh but stops after.

Reply:
> Thanks for your feedback. I'm not sure how to get the functionality of the like button correct. It works on initial load and on refresh but stops after.

Now it works fine